CMakeLists.txt

```diff
@@ -1,84 +1,118 @@
-cmake_minimum_required(VERSION 2.8.3)
+cmake_minimum_required(VERSION 3.5)
 project(urg_node)
 
-find_package(catkin REQUIRED COMPONENTS
-  diagnostic_updater dynamic_reconfigure laser_proc message_generation
-  nodelet rosconsole roscpp sensor_msgs std_msgs std_srvs tf urg_c
-)
-set_directory_properties(PROPERTIES COMPILE_OPTIONS "-std=c++11")
-
-# Dynamic reconfigure support
-generate_dynamic_reconfigure_options(cfg/URG.cfg)
-
-add_message_files(FILES
-  Status.msg
-)
+# Default to C++14
+if(NOT CMAKE_CXX_STANDARD)
+  set(CMAKE_CXX_STANDARD 14)
+endif()
+if(CMAKE_COMPILER_IS_GNUCXX OR CMAKE_CXX_COMPILER_ID MATCHES "Clang")
+  add_compile_options(-Wall -Wextra -Wpedantic)
+endif()
 
-generate_messages(
-  DEPENDENCIES std_msgs
+find_package(ament_cmake REQUIRED)
+find_package(urg_node_msgs REQUIRED)
+find_package(diagnostic_updater REQUIRED)
+#find_package(dynamic_reconfigure REQUIRED)
+find_package(laser_proc REQUIRED)
+find_package(rclcpp REQUIRED)
+find_package(sensor_msgs REQUIRED)
+find_package(std_msgs REQUIRED)
+find_package(std_srvs REQUIRED)
+find_package(tf2 REQUIRED)
+find_package(urg_c REQUIRED)
+find_package(builtin_interfaces REQUIRED)
+find_package(rosidl_default_generators REQUIRED)
+find_package(Boost REQUIRED thread)
```

Comment:
As far as I can tell, we are no longer using `boost` at all in this PR.  If that is the case, please remove this (and any references in the package.xml).

Reply:
boost is still being used for CRC. I don't have a simple STL solution for this. Hence, I am leaving it in.